packages/node/src/integrations/tracing/nest/sentry-nest-instrumentation.ts

```diff
@@ -163,33 +163,84 @@ export class SentryNestInstrumentation extends InstrumentationBase {
 
             target.prototype.intercept = new Proxy(target.prototype.intercept, {
               apply: (originalIntercept, thisArgIntercept, argsIntercept) => {
-                const [executionContext, next, args] = argsIntercept;
-                const prevSpan = getActiveSpan();
+                const context: MinimalNestJsExecutionContext = argsIntercept[0];
```

Comment:
h: If the context is different for each incoming request we can use the context directly to store the flag instead of storing it on the request object.

Reply:
Seems hard to verify and request to me seems intuitively more appropriate since that's exactly the granularity we want. What benefit would you hope to get from this?

Reply:
> What benefit would you hope to get from this?

Less complexity and more robustness by not calling type casted API. Would you mind testing whether the identity of the context changes for different requests?

Reply:
Seems to be okay so updated the implementation to write the property directly to the context object.